### PR TITLE
Add GCP CIS v2.0.0

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: cis_v200/docs/cis_v200_1_13.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## Description
+
```
Comment (khushboo9024) [inline cis_v200/docs/cis_v200_1_13.md:2]:
```suggestion
```

File: cis_v200/docs/cis_v200_1_13.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+### From Console
+### *Leaving Keys in Place*
+
+1. Go to `APIs & Services\Credentials` using [https://console.cloud.google.com/apis/credentials](https://console.cloud.google.com/apis/credentials)
+2. In the section `API Keys`, Click the `API Key Name`. The API Key properties display on a new page.
+3. In the `Key restrictions` section, set the application restrictions to any of `HTTP referrers, IP addresses, Android apps, iOS apps`.
+4. Click `Save`.
+5. Repeat steps 2,3,4 for every unrestricted API key.
+
+    **Note**: Do not set `HTTP referrers` to wild-cards (* or *.[TLD] or .[TLD]/) allowing access to any/wide HTTP referrer(s) Do not set `IP addresse`s and referrer to `any host (0.0.0.0 or 0.0.0.0/0 or ::0)`
```
Comment (khushboo9024) [inline cis_v200/docs/cis_v200_1_13.md:27]:
```suggestion
**Note**: Do not set `HTTP referrers` to wild-cards (* or *.[TLD] or .[TLD]/) allowing access to any/wide HTTP referrer(s) Do not set `IP addresse`s and referrer to `any host (0.0.0.0 or 0.0.0.0/0 or ::0)`
```

File: cis_v200/docs/cis_v200_2_1.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Cloud Audit Logging maintains two audit logs for each project, folder, and organization: Admin Activity and Data Access.
+
+1. Admin Activity logs contain log entries for API calls or other administrative actions that modify the configuration or metadata of resources. Admin Activity audit logs are enabled for all services and cannot be configured.
+2. Data Access audit logs record API calls that create, modify, or read user-provided data. These are disabled by default and should be enabled.
+
+There are three kinds of Data Access audit log information:
+   - Admin read: Records operations that read metadata or configuration information. Admin activity audit logs record writes of metadata and configuration information that cannot be disabled.
+   - Data read: Records operations that read user-provided data.
+   - Data write: Records operations that write user-provided data.
+
```
Comment (khushboo9024) [inline cis_v200/docs/cis_v200_2_1.md:14]:
```suggestion
 - Admin read: Records operations that read metadata or configuration information. Admin activity audit logs record writes of metadata and configuration information that cannot be disabled.
 - Data read: Records operations that read user-provided data.
 - Data write: Records operations that write user-provided data.

```

File: query/sql/sql_instance_postgresql_log_statement_database_flag_set_appropriately.sql [hunk 4]
```diff
@@ -0,0 +1,22 @@
+select
```
Comment (khushboo9024) [inline query/sql/sql_instance_postgresql_log_statement_database_flag_set_appropriately.sql:1]: Change query name to `sql_instance_postgresql_log_statement_database_flag_ddl.sql`

File: query/sql/sql_instance_postgresql_log_statement_database_flag_set_appropriately.sql [hunk 5]
```diff
@@ -0,0 +1,22 @@
+select 
+  -- Required Columns
+  self_link resource,
+  case
+    when database_version not like 'POSTGRES%' then 'skip'
+    when database_flags is null or not (database_flags @> '[{"name":"log_statement"}]') then 'alarm'
+    when database_flags @> '[{"name": "log_statement","value":"ddl"}]' then 'ok'
+    else 'alarm'
+  end status,
```
Comment (khushboo9024) [inline query/sql/sql_instance_postgresql_log_statement_database_flag_set_appropriately.sql:9]:
```suggestion
  end as status,
```
Please make this change to all places.

File: query/sql/sql_instance_postgresql_log_statement_database_flag_set_appropriately.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  case
+    when database_version not like 'POSTGRES%' then title || ' not a PostgreSQL database.'
+    when database_flags is null or not (database_flags @> '[{"name":"log_statement"}]') then title || ' log_statement database flag not set.'
+    when database_flags @> '[{"name": "log_statement","value":"ddl"}]' then title || ' log_statement database flag set to ddl.'
+    when database_flags @> '[{"name": "log_statement","value":"mod"}]' then title || ' log_statement database flag set to mod.'
+    when database_flags @> '[{"name": "log_statement","value":"all"}]' then title || ' log_statement database flag set to all.'
+    when database_flags @> '[{"name": "log_statement","value":"none"}]' then title || ' log_statement database flag set to none.'
+  end reason,
```
Comment (khushboo9024) [inline query/sql/sql_instance_postgresql_log_statement_database_flag_set_appropriately.sql:17]:
```suggestion
  end as reason,
```
Please make this change to all places.

File: query/sql/sql_instance_postgresql_log_min_messages_database_flag_set_minimum.sql [hunk 7]
```diff
@@ -0,0 +1,21 @@
+select
```
Comment (khushboo9024) [inline query/sql/sql_instance_postgresql_log_min_messages_database_flag_set_minimum.sql:1]: Change query name to `sql_instance_postgresql_log_min_messages_database_flag_error.sql`

File: query/sql/sql_instance_postgresql_log_error_verbosity_database_flag_set_default_or_stricter.sql [hunk 8]
```diff
@@ -0,0 +1,20 @@
+select 
+  -- Required Columns
+  self_link resource,
```
Comment (khushboo9024) [inline query/sql/sql_instance_postgresql_log_error_verbosity_database_flag_set_default_or_stricter.sql:3]: Change query name to `sql_instance_postgresql_log_error_verbosity_database_flag_default_or_stricter.sql`

File: query/logging/logging_https_load_balancer_enabled.sql [hunk 9]
```diff
@@ -0,0 +1,17 @@
+select
```
[the remark this answers is not in the document]
Comment (khushboo9024) [reply query/logging/logging_https_load_balancer_enabled.sql:1]: Please replace this query with below query
```
select
  -- Required Columns
  m.self_link as resource,
  case
    when s.self_link is null then 'skip'
    when s.log_config_enable then 'ok'
    else 'alarm'
  end status,
  case
    when s.self_link is null then m.name || ' uses backend bucket.'
    when s.log_config_enable then m.name || ' logging enabled.'
    else m.name || ' logging disabled.'
  end reason,
  -- Additional Dimensions
  m.location,
  m.project
from
  gcp_compute_url_map as m
  left join gcp_compute_backend_service as s on s.self_link = m.default_service;
  ```
  Also move this query to compute folder and change the query name to `compute_https_load_balancer_logging_enabled.sql`

File: cis_v200/section_1.sp [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+
+control "cis_v200_1_13" {
+  title         = "1.13 Ensure API keys are restricted to use by only specified Hosts and Apps"
+  description   = "Unrestricted keys are insecure because they can be viewed publicly, such as from within a browser, or they can be accessed on a device where the key resides. It is recommended to restrict API key usage to trusted hosts, HTTP referrers and apps."
+  query         = query.manual_control
+  documentation = file("./cis_v200/docs/cis_v200_1_13.md")
+
+  tags = merge(local.cis_v200_1_common_tags, {
+    cis_item_id = "1.13"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:212]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_1.sp [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+
+control "cis_v200_1_14" {
+  title         = "1.14 Ensure API keys are restricted to only APIs that application needs access"
+  description   = "API keys are insecure because they can be viewed publicly, such as from within a browser, or they can be accessed on a device where the key resides. It is recommended to restrict API keys to use (call) only APIs required by an application."
+  query         = query.manual_control
+  documentation = file("./cis_v200/docs/cis_v200_1_14.md")
+
+  tags = merge(local.cis_v200_1_common_tags, {
+    cis_item_id = "1.14"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:226]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_1.sp [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+    cis_type    = "manual"
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:227]:
```suggestion
    cis_type    = "automated"
```

File: cis_v200/section_1.sp [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+
+control "cis_v200_1_15" {
+  title         = "1.15 Ensure API keys are rotated every 90 days"
+  description   = "It is recommended to rotate API keys every 90 days."
+  query         = query.manual_control
+  documentation = file("./cis_v200/docs/cis_v200_1_15.md")
+
+  tags = merge(local.cis_v200_1_common_tags, {
+    cis_item_id = "1.15"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:240]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_1.sp [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+    cis_type    = "manual"
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:241]:
```suggestion
    cis_type    = "automated"
```

File: cis_v200/section_2.sp [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+
+control "cis_v200_2_3" {
+  title         = "2.3 Ensure that retention policies on log buckets are configured using Bucket Lock"
+  description   = "Enabling retention policies on log buckets will protect logs stored in cloud storage buckets from being overwritten or accidentally deleted. It is recommended to set up retention policies and configure Bucket Lock on all storage buckets that are used as log sinks."
+  query         = query.logging_bucket_retention_policy_enabled
+  documentation = file("./cis_v200/docs/cis_v200_2_3.md")
+
+  tags = merge(local.cis_v200_2_common_tags, {
+    cis_item_id = "2.3"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_2.sp:70]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_2.sp [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+
+control "cis_v200_2_7" {
+  title         = "2.7 Ensure that the log metric filter and alerts exist for VPC Network Firewall rule changes"
+  description   = "It is recommended that a metric filter and alarm be established for Virtual Private Cloud (VPC) Network Firewall rule changes."
+  query         = query.logging_metric_alert_firewall_rule_changes
+  documentation = file("./cis_v200/docs/cis_v200_2_7.md")
+
+  tags = merge(local.cis_v200_2_common_tags, {
+    cis_item_id = "2.7"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_2.sp:126]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_2.sp [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+
+control "cis_v200_2_8" {
+  title         = "2.8 Ensure that the log metric filter and alerts exist for VPC network route changes"
+  description   = "It is recommended that a metric filter and alarm be established for Virtual Private Cloud (VPC) network route changes."
+  query         = query.logging_metric_alert_network_route_changes
+  documentation = file("./cis_v200/docs/cis_v200_2_8.md")
+
+  tags = merge(local.cis_v200_2_common_tags, {
+    cis_item_id = "2.8"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_2.sp:140]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_2.sp [hunk 18]
```diff
@@ -0,0 +1,256 @@
...
+
+control "cis_v200_2_9" {
+  title         = "2.9 Ensure that the log metric filter and alerts exist for VPC network changes"
+  description   = "It is recommended that a metric filter and alarm be established for Virtual Private Cloud (VPC) network changes."
+  query         = query.logging_metric_alert_network_changes
+  documentation = file("./cis_v200/docs/cis_v200_2_9.md")
+
+  tags = merge(local.cis_v200_2_common_tags, {
+    cis_item_id = "2.9"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_2.sp:154]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_2.sp [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+
+control "cis_v200_2_10" {
+  title         = "2.10 Ensure that the log metric filter and alerts exist for Cloud Storage IAM permission changes"
+  description   = "It is recommended that a metric filter and alarm be established for Cloud Storage Bucket IAM changes."
+  query         = query.logging_metric_alert_storage_iam_permission_changes
+  documentation = file("./cis_v200/docs/cis_v200_2_10.md")
+
+  tags = merge(local.cis_v200_2_common_tags, {
+    cis_item_id = "2.10"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_2.sp:168]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/section_2.sp [hunk 20]
```diff
@@ -0,0 +1,256 @@
...
+
+control "cis_v200_2_11" {
+  title         = "2.11 Ensure that the log metric filter and alerts exist for SQL instance configuration changes"
+  description   = "It is recommended that a metric filter and alarm be established for SQL instance configuration changes."
+  query         = query.logging_metric_alert_sql_instance_configuration_changes
+  documentation = file("./cis_v200/docs/cis_v200_2_11.md")
+
+  tags = merge(local.cis_v200_2_common_tags, {
+    cis_item_id = "2.11"
+    cis_level   = "1"
```
Comment (khushboo9024) [inline cis_v200/section_2.sp:182]:
```suggestion
    cis_level   = "2"
```

File: cis_v200/docs/cis_v200_4_6.md [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+## Remediation
+
+You only edit the `canIpForward` setting at instance creation time. Therefore, you need to delete the instance and create a new one where `canIpForward` is set to `false`.
+
+### From Google Cloud Console
+
+1. Go to the `VM Instances` page by visiting: [https://console.cloud.google.com/compute/instances](https://console.cloud.google.com/compute/instances).
+2. Select the `VM Instance` you want to remediate.
+3. Click the `Delete` button.
+4. On the 'VM Instances' page, click `CREATE INSTANCE'.
```
Comment (khushboo9024) [inline cis_v200/docs/cis_v200_4_6.md:18]:
```suggestion
4. On the 'VM Instances' page, click `CREATE INSTANCE`.
```

File: cis_v200/docs/cis_v200_4_12.md [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+
+The output should show a list of packages that have updates available.
+
+Windows
+```bash
+ping http://windowsupdate.microsoft.com/
+```
+
+The ping should successfully be delivered and received.
+
```
Comment (khushboo9024) [inline cis_v200/docs/cis_v200_4_12.md:90]: Remove this extra line

File: query/sql/sql_instance_postgresql_log_min_messages_database_flag_error.sql [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+    when database_flags is null or not (database_flags @> '[{"name":"log_min_messages"}]') then 'alarm'
+    when database_flags @> '[{"name":"log_min_messages","value":"error"}]' or database_flags @> '[{"name":"log_min_messages","value":"warning"}]' then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when database_version not like 'POSTGRES%' then title || ' not a PostgreSQL database.'
+    when database_flags is null or not (database_flags @> '[{"name":"log_min_messages"}]') then title || ' log_min_messages database flag not set.'
+    when database_flags @> '[{"name":"log_min_messages","value":"error"}]' then title || ' log_min_messages database flag set to ERROR.'
+    when database_flags @> '[{"name":"log_min_messages","value":"warning"}]' then title || ' log_min_messages database flag set to WARNING.'
+    else  title || ' log_min_messages database flag not set at minimum to WARNING.'
```
Comment (khushboo9024) [inline query/sql/sql_instance_postgresql_log_min_messages_database_flag_error.sql:15]:
```suggestion
    else title || ' log_min_messages database flag not set at minimum to WARNING.'
```

File: query/sql/sql_instance_not_publicly_accessible.sql [hunk 24]
```diff
@@ -0,0 +1,16 @@
+select 
+  -- Required Columns
+  self_link resource,
+  case
+    when (ip_addresses @> '[{"type": "PRIVATE"}]' and ip_configuration ->> 'privateNetwork' is not null) and not (ip_addresses @> '[{"type": "PRIMARY"}]') then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when (ip_addresses @> '[{"type": "PRIVATE"}]' and ip_configuration ->> 'privateNetwork' is not null) and not (ip_addresses @> '[{"type": "PRIMARY"}]')  then title || ' not publicly accessible.'
```
Comment (khushboo9024) [inline query/sql/sql_instance_not_publicly_accessible.sql:9]:
```suggestion
    when (ip_addresses @> '[{"type": "PRIVATE"}]' and ip_configuration ->> 'privateNetwork' is not null) and not (ip_addresses @> '[{"type": "PRIMARY"}]') then title || ' not publicly accessible.'
```

**Review by khushboo9024 — APPROVED** [review]

**Review by misraved — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please take a look at the review comments.

Please ensure that spacing between the lines in the docs is uniform.

**Review by misraved — APPROVED** [review]

**Pull request merged**